adapter/config/crd/controller.go

```diff
@@ -47,6 +47,10 @@ type cacheHandler struct {
 // NewController creates a new Kubernetes controller for CRDs
 // Use "" for namespace to listen for all namespace changes
 func NewController(client *Client, options kube.ControllerOptions) model.ConfigStoreCache {
+
+	glog.V(2).Infof("CRD controller running in namespace %s, watching app namespaces %s",
+		options.Namespace, options.AppNamespace)
```

Comment:
maybe you should remove Namespace entirely and only have AppNamespace and IstioNamespace ?

Reply:
namespace is a nice name to indicate where the pod is deployed and where container is running, so I will not rename.

Reply:
see my other question/request about commenting more which is which in controller.go, also not sure your explanation (where the pod is deployed) matches the comment above "Use "" for namespace to listen for all namespace changes"

Reply:
Why does controller need to know which namespace it runs in? I should be able to run it out of cluster.

Reply:
Indeed, it does not need to know, but in the overall code I had seen the pattern of using downward API and passing POD_NAMESPACE, which then gets translated into options.Namespace, for all controllers.